ports/raspberrypi/common-hal/countio/Counter.c

```diff
@@ -0,0 +1,81 @@
+#include "common-hal/countio/Counter.h"
+
+#include "py/runtime.h"
+#include "py/mpstate.h"
+#include "supervisor/shared/translate.h"
+
+#include "src/rp2_common/hardware_gpio/include/hardware/gpio.h"
+#include "src/rp2_common/hardware_pwm/include/hardware/pwm.h"
+#include "src/rp2_common/hardware_irq/include/hardware/irq.h"
+
+
+void common_hal_countio_counter_construct(countio_counter_obj_t* self,
+    const mcu_pin_obj_t* pin_a) {
+
+    if (pwm_gpio_to_channel(pin_a->number) != PWM_CHAN_B) {
+        mp_raise_RuntimeError(translate("Pin must be on PWM Channel B"));
+    }
+
+    self->pin_a = pin_a->number;
+    self->slice_num = pwm_gpio_to_slice_num(self->pin_a);
+
+    pwm_clear_irq(self->slice_num);
+    pwm_set_irq_enabled(self->slice_num, true);
+    irq_set_exclusive_handler(PWM_IRQ_WRAP, counter_interrupt_handler);
+    irq_set_enabled(PWM_IRQ_WRAP, true);
+
+    pwm_config cfg = pwm_get_default_config();
+    pwm_config_set_clkdiv_mode(&cfg, PWM_DIV_B_RISING);
+    pwm_init(self->slice_num, &cfg, false);
+    gpio_set_function(self->pin_a, GPIO_FUNC_PWM);
+
+    self->count = 0;
+
+    claim_pin(pin_a);
+
+    MP_STATE_PORT(counting) = self;
```

Comment:
This won't work if you have multiple counters going at once. I think you want an array of size 8 (one for each slice) that can store each object. Then in the interrupt you'll need to check which slice wrapped and increment the corresponding counters count.

Reply:
Fixed in the last commit